### PR TITLE
ARMClientCache should only check the namespace the resource is in

### PR DESCRIPTION
**Pull request by matthchr**
No cross namespace secret references should be allowed as there is a risk for escalation of privilege.

This is a breaking change, but unfortunately a required one from a security perspective.

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

**Comment by matthchr** [comment]
I've also updated my documentation PR to talk about security best practices: #2915

File: v2/internal/reconcilers/arm/arm_client_cache.go [hunk 1]
```diff
@@ -166,13 +175,7 @@ func (c *ARMClientCache) getCredentialFromAnnotation(ctx context.Context, obj ge
 }
 
 func getSecretNameFromAnnotation(credentialFrom string, resourceNamespace string) types.NamespacedName {
```
Comment (super-harsh) [inline v2/internal/reconcilers/arm/arm_client_cache.go:177]: Now we can totally get rid of this method and just use `types.NamespacedName{Namespace: resourceNamespace, Name: credentialFrom}` instead?

Comment (matthchr) [reply v2/internal/reconcilers/arm/arm_client_cache.go:177]: Fixed

**Review by super-harsh — APPROVED** [review]

**Pull request merged**